### PR TITLE
DocParser: Improve private typehints

### PR DESCRIPTION
**Pull request by jkufner**
- Improve private typehints and annotations; unify imports
- syntaxError() returns the exception rather than throwing it; it makes the code more readable and IDE less confused

None of these changes should change any behavior.

### CODE REVIEW

**Comment by greg0ire** [comment]
Blocked by #341

**Comment by greg0ire** [comment]
Thanks @jkufner !

File: lib/Doctrine/Common/Annotations/DocParser.php [hunk 1]
```diff
@@ -358,10 +363,8 @@ public function parse($input, $context = '')
      * Finds the first valid annotation
      *
      * @param string $input The docblock string to parse
-     *
-     * @return int|null
      */
-    private function findInitialTokenPosition($input)
+    private function findInitialTokenPosition($input): ?int
```
Comment (stof) [inline lib/Doctrine/Common/Annotations/DocParser.php:367]: if you add the return type, why not adding the argument type too ?

File: lib/Doctrine/Common/Annotations/DocParser.php [hunk 2]
```diff
@@ -421,12 +423,8 @@ private function matchAny(array $tokens)
      *
      * @param string     $expected Expected string.
      * @param array|null $token    Optional token.
```
Comment (stof) [inline lib/Doctrine/Common/Annotations/DocParser.php:425]: the descriptions don't really provide any value here, and the types are not specified through typehints

**Pull request merged**